shared/tests/utils/test_mcp_utils.py

```
@@ -29,10 +29,110 @@ def test_replace_mcp_server_variables_replaces_backend_url_and_task_token():
+        "git_repo": "myorg/myrepo",
+        "git_domain": "github.com",
+        "branch_name": "feature/test",
+        "bot": [],
+    }
+
+    replaced = replace_mcp_server_variables(mcp_servers, task_data)
+
+    assert replaced["git-server"]["args"][3] == "ghp_abc123"
+    assert replaced["git-server"]["args"][5] == "myorg/myrepo"
+    assert replaced["git-server"]["args"][7] == "feature/test"
+    assert replaced["git-server"]["env"]["GIT_TOKEN"] == "ghp_abc123"
```

Ruff (0.15.2)

[error] Possible hardcoded password assigned to: "GIT_TOKEN"

(S105)